app/controllers/api/v1/move_events_controller.rb

```diff
@@ -59,39 +75,28 @@ def timestamp
         @timestamp ||= Time.zone.parse(event_params.dig(:attributes, :timestamp))
       end
 
-      def notes
-        @notes ||= event_params.dig(:attributes, :notes)
-      end
-
-      def to_location
-        @to_location ||= Location.find(event_params.dig(:relationships, :to_location, :data, :id))
+      def supplier_id
+        # NB: not all events will have a supplier_id so this could well be nil
+        current_user.owner&.id
       end
 
       def move
-        @move ||= Move.accessible_by(current_ability).find(params.require(:move_id))
+        @move ||= Move.accessible_by(current_ability).find(params.require(:id))
```

Comment:
Is this as a result of amending the routes structure? Might be worth double checking that this is the correct id (run `rails routes`) - I would expect this to still be named `:id` from the URL params.

Reply:
yes, it is the routes have changed: `events` used to be a nested resource inside of `/moves/{move_id}/events`, so it was necessary for rails to distinguish from move_id and (id = event_id). But now the events endpoint is just an additional method on a move resource, rather than a resource in its own right..